src/main/java/org/broadinstitute/hellbender/utils/GATKProtectedVariantContextUtils.java

```diff
@@ -523,4 +523,217 @@ private static int getMinBaseQualityForAlleleInRead(final PileupElement pileupEl
+        return reference.length() > alternate.length();
+    }
+
+    /**
+     * Determines whether the given reference and alternate alleles constitute an insertion or deletion mutation.
+     * @param reference A {@link String} containing the bases of the reference allele.  Must not be {@code null}.
+     * @param alternate A {@link String} containing the bases of the alternate / variant allele.  Must not be {@code null}.
+     * @return {@code true} if replacing the reference with the alternate results in an insertion or deletion.  {@code false} otherwise.
+     */
+    public static boolean isIndel(final String reference, final String alternate) {
+
+        Utils.nonNull(reference);
+        Utils.nonNull(alternate);
+
+        if ( reference.contains("*") ) {
+            throw new GATKException("Allele strings must contain only valid bases (" + new String(BaseUtils.BASE_CHARS) + ").  Found: " + reference);
+        }
+        else if ( alternate.contains("*") ) {
+            throw new GATKException("Allele strings must contain only valid bases (" + new String(BaseUtils.BASE_CHARS) + ").  Found: " + alternate);
+        }
+
+        // If we do not have the same number of bases in the reference and alternate alleles,
+        // then we have an indel:
+        return reference.length() != alternate.length();
+    }
+
+    /**
+     * Determines whether the given reference and alternate alleles constitute an insertion or deletion mutation.
+     * @param reference The reference {@link Allele}.  Must not be {@code null}.
+     * @param alternate The alternate / variant {@link Allele}.  Must not be {@code null}.
+     * @return {@code true} if replacing the reference with the alternate results in an insertion or deletion.  {@code false} otherwise.
+     */
+    public static boolean isIndel(final Allele reference, final Allele alternate) {
+
+        Utils.nonNull(reference);
+        Utils.nonNull(alternate);
+
+        // If we do not have the same number of bases in the reference and alternate alleles,
+        // then we have an indel:
+        return reference.length() != alternate.length();
+    }
+
+    /**
+     * Determines whether the given reference and alternate alleles constitute an Oligo-Nucleotide Polymorphism (ONP).
+     * For Funcotator purposes, an ONP is a mutation/variant that is a substitution of one or more consecutive bases.
+     * @param reference The reference {@link Allele}.  Must not be {@code null}.
+     * @param alternate The alternate / variant {@link Allele}.  Must not be {@code null}.
+     * @return {@code true} if replacing the reference with the alternate results in an ONP.  {@code false} otherwise.
+     */
+    public static boolean isOnp(final Allele reference, final Allele alternate) {
+
+        Utils.nonNull(reference);
+        Utils.nonNull(alternate);
+
+        // If we have more bases in the alternate, we have an insertion:
+        return reference.length() == alternate.length();
```

Comment:
You may want to check for duplication of bases.  It's not so much that the length is equal, but that the length is equal and that they differ at more than one consecutive bases.

Please be careful about co-opting ONP to mean one or more, since that is not the definition...  I usually use `xNP` for that.

Reply:
Just fixed the check.

Are they not the same?  I can change my documentation to XNP.